provider/source/src/currency/displayname.rs

```diff
@@ -0,0 +1,114 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::cldr_serde;
+use crate::SourceDataProvider;
+use icu::experimental::dimension::provider::currency_displayname::*;
+use icu_provider::prelude::*;
+use std::borrow::Cow;
+use std::collections::HashSet;
+
+impl DataProvider<CurrencyDisplaynameV1Marker> for crate::SourceDataProvider {
+    fn load(
+        &self,
+        req: DataRequest,
+    ) -> Result<DataResponse<CurrencyDisplaynameV1Marker>, DataError> {
+        self.check_req::<CurrencyDisplaynameV1Marker>(req)?;
+
+        let currencies_resource: &cldr_serde::currencies::data::Resource =
+            self.cldr()?
+                .numbers()
+                .read_and_parse(req.id.locale, "currencies.json")?;
+
+        let currency = currencies_resource
+            .main
+            .value
+            .numbers
+            .currencies
+            .get(req.id.marker_attributes.as_str())
+            .ok_or(DataError::custom("No currency associated with the aux key"))?;
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: DataPayload::from_owned(CurrencyDisplaynameV1 {
+                display_name: Cow::Owned(
+                    currency
+                        .display_name
+                        .as_deref()
+                        .or(currency.one.as_deref())
+                        .ok_or(DataError::custom(
+                            "No display name or `one` count found for the currency",
+                        ))?
```

Comment:
This should be `DataErrorKind::IdentifierNotFound`.

Reply:
done